static/js/message_edit.js

```diff
@@ -212,9 +212,30 @@ function handle_edit_keydown(from_topic_edited_only, e) {
     var row;
     var code = e.keyCode || e.which;
 
-    if ($(e.target).hasClass("message_edit_content") && code === 13 &&
-        (e.metaKey || e.ctrlKey)) {
-        row = $(".message_edit_content").filter(":focus").closest(".message_row");
+    if ($(e.target).hasClass("message_edit_content") && code === 13) {
+        var has_non_shift_modifier_key = e.ctrlKey || e.metaKey || e.altKey;
+        var has_modifier_key = e.shiftKey || has_non_shift_modifier_key;
+        var this_enter_edits;
+        if (page_params.enter_sends) {
+            this_enter_edits = !has_modifier_key;
+        } else {
+            this_enter_edits = has_non_shift_modifier_key;
+        }
+        if (this_enter_edits) {
+            row = $(".message_edit_content").filter(":focus").closest(".message_row");
```

Comment:
The first half of this should instead be extracted to a reusable function in `static/js/composebox_typeahead.js`, maybe `composebox_typeahead.should_enter_send(event)` -- it's crazy to copy this complicated logic without the comments.